### DISCUSSION TITLE
Usage of declarative shadom DOM

### DISCUSSION THREAD

**Discussion by edimoldovan** [nuejs/nue#44] [Ideas]
Would be interesting to see if neujs can compile its components to web components so their ability to automatically hydrate SSR web components and their shadow DOM to create a resumable app independent from the backend stack.

**Comment by tipiirai** [comment]
I have been thinking about this. Some discussion about webcomponents here: 

https://github.com/nuejs/nuejs/discussions/1#discussioncomment-7053333

One challenge with that approach is that how to provide complex data (arrays, objects,...) for the component as well as the dependencies?

**Comment by edimoldovan** [comment]
The how of that is indeed the million dollar question -- I haven't yet figured it out myself either. Maybe something around linking DOM elements to model, something like data-model attribute and then children would have fields mapped in a similar way with data-field. Just an idea I had, haven't tried it yet.
